dev-client/src/components/siteNotes/EditSiteNoteScreen.tsx

```diff
@@ -0,0 +1,150 @@
+import {SITE_NOTE_MIN_LENGTH} from 'terraso-mobile-client/constants';
+import {ScreenScaffold} from 'terraso-mobile-client/screens/ScreenScaffold';
+import {useNavigation} from 'terraso-mobile-client/screens/AppScaffold';
+import {Keyboard, KeyboardAvoidingView, Platform} from 'react-native';
+
+type Props = {
+  note: SiteNote;
+};
+
+export const EditSiteNoteScreen = ({note}: Props) => {
+  const {t} = useTranslation();
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
+
+  const notesFormSchema = yup.object().shape({
+    content: yup.string().required(
+      t('site.notes.min_length_error', {
+        min: SITE_NOTE_MIN_LENGTH,
+      }),
+    ),
+  });
+
+  const handleUpdateNote = async (content: string) => {
+    if (!content.trim()) {
+      return;
+    }
+    Keyboard.dismiss();
+    try {
+      const siteNoteInput: SiteNoteUpdateMutationInput = {
+        id: note.id,
+        content: content,
+      };
+      const result = await dispatch(updateSiteNote(siteNoteInput));
+      if (result.payload && 'error' in result.payload) {
+        console.error(result.payload.error);
+        console.error(result.payload.parsedErrors);
+      }
+    } catch (error) {
+      console.error('Failed to update note:', error);
+    } finally {
+      navigation.pop();
+    }
+  };
+
+  const handleDelete = useCallback(
+    async (setSubmitting: (isSubmitting: boolean) => void) => {
+      setSubmitting(true);
+      await dispatch(deleteSiteNote(note)).then(() => navigation.pop());
+      setSubmitting(false);
+    },
+    [navigation, dispatch, note],
+  );
+
+  return (
+    <ScreenScaffold BottomNavigation={null} AppBar={null}>
+      <KeyboardAvoidingView
+        behavior={Platform.OS === 'ios' ? 'padding' : 'height'} // Handle platform-specific keyboard avoidance
+        // eslint-disable-next-line react-native/no-inline-styles
+        style={{flex: 1}}>
+        <Box pt={10} pl={5} pr={5} pb={10} flex={1}>
+          <Formik
+            initialValues={{content: note.content}}
+            validationSchema={notesFormSchema}
+            onSubmit={async (values, actions) => {
+              actions.setSubmitting(true);
+              await handleUpdateNote(values.content).then(() =>
+                actions.setSubmitting(false),
+              );
+            }}>
+            {formikProps => {
+              const {handleSubmit, isSubmitting, setSubmitting, values} =
+                formikProps;
+
+              return (
+                <>
+                  <Heading variant="h6" pb={7}>
+                    {t('site.notes.add_title')}
+                  </Heading>
+                  <Box flexGrow={1}>
+                    <SiteNoteForm
+                      content={formikProps.values.content}
+                      onChangeContent={formikProps.handleChange('content')}
+                      onBlurContent={formikProps.handleBlur('content')}
```

Comment:
comment: I think the handlers here might be already used by the `FormWrapper` component.